File: padroesEstruturais/padrao-de-projeto-adapter/php/src/Adapter/DanoAdapter.php
<?php

namespace Adapter\Php\Adapter;

class DanoAdapter extends DanoPorcentagem
{
    private IDano $danoNumerico;

    public function __construct(IDano $danoNumerico)
    {
        $this->danoNumerico = $danoNumerico;
    }

    public function recuperarDano(): String
    {
        if($this->getDano() === "100%") {
            return $this->danoNumerico->setDano(1000)->getDano();
        }

        if($this->getDano() === "50%") {
            return $this->danoNumerico->setDano(500)->getDano();
        }

        if($this->getDano() === "10%") {
            return $this->danoNumerico->setDano(100)->getDano();
        }

        return $this->danoNumerico->getDano();
    }

    public function salvarDano(): DanoAdapter
    {
        if($this->danoNumerico->getDano() >= 1000) {
            $this->setDano("100%");
            return $this;
        }

        if($this->danoNumerico->getDano() >= 500) {
            $this->setDano("50%+");
            return $this;
        }

        if($this->danoNumerico->getDano() >= 100) {
            $this->setDano("10%+");
            return $this;
        }

        return $this;
    }
}

File: padroesEstruturais/padrao-de-projeto-adapter/java/src/main/java/DanoAdapter.java
public class DanoAdapter extends DanoPorcentagem {

    IDano avaliacaoNumerico = null;

    public DanoAdapter(IDano avaliacaoNumerico) {
        this.avaliacaoNumerico = avaliacaoNumerico;
    }

    public float recuperaDano() {
        if(this.getDano().equals("100%")) {
            return (this.avaliacaoNumerico.setDano(1000f)).getDano();
        }

        if(this.getDano().equals("50%+")) {
            return (this.avaliacaoNumerico.setDano(500f)).getDano();
        }

        if(this.getDano().equals("10%+")) {
            return (this.avaliacaoNumerico.setDano(100f)).getDano();
        }

        return this.avaliacaoNumerico.getDano();
    }

    public DanoAdapter salvarDano() {
        if(this.avaliacaoNumerico.getDano() >= 1000f) {
            this.setDano("100%");
            return this;
        }

        if(this.avaliacaoNumerico.getDano() >= 500f) {
            this.setDano("50%+");
            return this;
        }

        if(this.avaliacaoNumerico.getDano() >= 100f) {
            this.setDano("10%+");
            return this;
        }

        return this;
    }

}
